Bump k8s v1.33

Vicente-Cheng opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as a draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
#### Before this PR:

if we use the k8s.io/client-go higher than v0.33 and kubevirt.io/client-go simultaneously, we will get the complaint as below:
```
vendor/kubevirt.io/client-go/kubecli/kubevirt_test_utils.go:31:9: cannot use MockKubevirtClientInstance (variable of type *MockKubevirtClient) as KubevirtClient value in return statement: *MockKubevirtClient does not implement KubevirtClient (missing method ResourceV1beta2)
```

#### After this PR:

Should not get the complaint because the `MockKubevirtClient` have the corresponding interface.

### References
- Fixes # https://github.com/kubevirt/kubevirt/issues/15717

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Bump k8s v1.33
```

fossedihelm commented:
@Vicente-Cheng Exactly :slightly_smiling_face:

Vicente-Cheng commented:
Thanks for the reminder! @fossedihelm 
I will rebase once the builder PR is merged.

RamLavi commented:
Hey, FYI we need this bump PR merged in order to bump KMP k8s.io as well.

Vicente-Cheng commented on go.mod:
```diff
@@ -1,6 +1,6 @@
 module kubevirt.io/kubevirt
 
-go 1.24
+go 1.24.0
```
Thanks for the hint @fossedihelm.
I try to remove the .z version, but after executing `make deps-sync`.
We automatically add them back.

Do you know how to avoid this?

Vicente-Cheng replied:
OK, looks like the `make deps-sync` would leverage the local Go version.
I skipped this part and ran the `make && make generate`
seems everything works well.

Updated.

fossedihelm replied:
Looks like convention is being changed: https://github.com/kubevirt/kubevirt/pull/15908
Apologies.

fossedihelm commented:
/test pull-kubevirt-build
/test pull-kubevirt-generate
/test pull-kubevirt-code-lint
/test pull-kubevirt-verify-go-mod
/test pull-kubevirt-unit-test
/test pull-kubevirt-manifests
/test pull-kubevirt-check-unassigned-tests

fossedihelm commented:
@Vicente-Cheng Can you fix the lint errors?
You can check with `make lint`.
Thanks

Vicente-Cheng commented:
> @Vicente-Cheng Can you fix the lint errors? You can check with `make lint`. Thanks

sure, let me do it.

Vicente-Cheng commented:
Hi @fossedihelm, 

Sorry for the delay.
fixed up the `make lint` and also ran the `make && make generate`  and `make test` locally

Please help to review again!

Vicente-Cheng commented:
> @Vicente-Cheng still missing some generated code

Hi @fossedihelm, 
Did you mean this error?
```
ERROR: git tree state is not clean!
You probably need to run 'make generate' or 'make' and commit the changes
On branch main
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   go.mod
	modified:   go.work
	modified:   staging/src/kubevirt.io/api/go.mod
	modified:   staging/src/kubevirt.io/client-go/go.mod
	modified:   vendor/modules.txt
no changes added to commit (use "git add" and/or "git commit -a")
diff --git a/go.mod b/go.mod
index d8e0b4b82b..da29730be3 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,8 @@
 module kubevirt.io/kubevirt
 
-go 1.24
+go 1.24.0
+
+toolchain go1.24.9
 
 require (
 	github.com/cheggaaa/pb/v3 v3.1.0
diff --git a/go.work b/go.work
index 65485ffdab..39092f9c54 100644
--- a/go.work
+++ b/go.work
@@ -1,4 +1,6 @@
-go 1.24
+go 1.24.0
+
+toolchain go1.24.9
 
 use (
 	.
diff --git a/staging/src/kubevirt.io/api/go.mod b/staging/src/kubevirt.io/api/go.mod
index 81d9e88d3d..65d69524f6 100644
--- a/staging/src/kubevirt.io/api/go.mod
+++ b/staging/src/kubevirt.io/api/go.mod
@@ -1,6 +1,8 @@
 module kubevirt.io/api
 
-go 1.24
+go 1.24.0
+
+toolchain go1.24.9
 
 require (
 	github.com/google/go-cmp v0.7.0
diff --git a/staging/src/kubevirt.io/client-go/go.mod b/staging/src/kubevirt.io/client-go/go.mod
index 23bf872d00..d50de44479 100644
--- a/staging/src/kubevirt.io/client-go/go.mod
+++ b/staging/src/kubevirt.io/client-go/go.mod
@@ -1,6 +1,8 @@
 module kubevirt.io/client-go
 
-go 1.24
+go 1.24.0
+
+toolchain go1.24.9
 
 require (
 	github.com/go-kit/log v0.2.1
diff --git a/vendor/modules.txt b/vendor/modules.txt
index aaf140e924..93b2a68e10 100[644](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/15718/pull-kubevirt-verify-go-mod/1980202603023175680#1:build-log.txt%3A644)
--- a/vendor/modules.txt
+++ b/vendor/modules.txt
@@ -1280,9 +1280,9 @@ k8s.io/utils/pointer
 k8s.io/utils/ptr
 k8s.io/utils/trace
 # kubevirt.io/api v0.0.0-00010101000000-000000000000 => ./staging/src/kubevirt.io/api
-## explicit; go 1.24.7
+## explicit; go 1.24.0
 # kubevirt.io/client-go v0.0.0-00010101000000-000000000000 => ./staging/src/kubevirt.io/client-go
-## explicit; go 1.24.7
+## explicit; go 1.24.0
 # kubevirt.io/containerized-data-importer-api v1.63.1
 ## explicit; go 1.23.0
 kubevirt.io/containerized-data-importer-api/pkg/apis/core
+ EXIT_VALUE=1
+ set +o xtrace
```

The above error might be related to the 'z' version applied.
I found we would also generate that on `make & make generate`.

OK.. seems I need to wait for this fix? https://github.com/kubevirt/kubevirt/pull/15908

Vicente-Cheng commented:
Hi @fossedihelm, 
I rebased the latest main and run the `make && make generate` locally.
It did not show anything that needs to be added.

Could you help trigger the ci again?

fossedihelm commented:
/test pull-kubevirt-build
/test pull-kubevirt-generate
/test pull-kubevirt-code-lint
/test pull-kubevirt-verify-go-mod
/test pull-kubevirt-unit-test
/test pull-kubevirt-manifests
/test pull-kubevirt-check-unassigned-tests

fossedihelm commented:
@Vicente-Cheng I tried something here https://github.com/kubevirt/kubevirt/pull/15941 and CI is green

Basically I:
- interactive rebase editing commit [`2de1ba4` (#15718)](https://github.com/kubevirt/kubevirt/pull/15718/commits/2de1ba44c473e432ade3090dd0670feb66b07241)
- `make deps-sync`
- drop the toolchain instruction from go.mod(s)
- `make && make generate`
- `git add -u`
- `git commint --amend --no-edit`
- `git rebase --continue`

After this I rebased on main.
Hope this helps
Thanks
